core/src/main/java/nl/nn/adapterframework/pipes/XmlSwitch.java

```diff
@@ -158,10 +158,9 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 					parametervalues = parameterList.getValues(message, session, isNamespaceAware()).getValueMap();
 				}
 				if(StringUtils.isNotEmpty(getSessionKey())) {
-					forward = transformerPool.transform(session.getMessage(getSessionKey()), parametervalues);
+					forward = transformerPool.transform(Message.asString(session.get(sessionKey)), parametervalues, isNamespaceAware());
 				} else {
-					message.preserve();
-					forward = transformerPool.transform(message, parametervalues);
+					forward = transformerPool.transform(message.asString(), parametervalues, isNamespaceAware());
```

Comment:
is het stiekem niet handig dit alleen te doen wanneer namespaceAware aanstaat?

Reply:
Zo is het wel gedaan in 7.8, maar dat wilde ik niet backporten

Reply:
als dit een simpele if is, doe het toch maar.
en ook de 7.7-release branch mergen